scripts/update_devices_db.py

```diff
@@ -0,0 +1,297 @@
+            page_assets = await response.json()
+
+            # If no more assets, break the loop
+            if not page_assets:
+                break
+
+            assets.extend(page_assets)
+            page += 1
+
+    return assets
+
+async def upload_release_asset(session, token, repo, asset_path, release_version=None):
+    """ Upload a release asset to GitHub """
+    print(f"Preparing to upload asset: {os.path.basename(asset_path)}...")
+    headers = {'Authorization': f'token {token}', 'Content-Type': 'application/octet-stream'}
+    release_url = f"https://api.github.com/repos/{repo}/releases/latest"
+    if release_version:
+        if len(release_version) and ('latest' != release_version):
+            release_url = f"https://api.github.com/repos/{repo}/releases/tags/{release_version}"
+    async with session.get(release_url, headers=headers) as response:
+        response_data = await response.json()
+        release_id = response_data['id']
+
+    # Get all assets for the release
+    assets = await get_all_assets(session, token, repo, release_id)
+    # Then, filter out the one needed
+    existing_asset = next((asset for asset in assets if asset['name'] == os.path.basename(asset_path)), None)
+
+    # If the asset exists, delete it
+    if existing_asset:
+        delete_url = existing_asset['url']
+        async with session.delete(delete_url, headers=headers) as response:
+            if response.status == 204:
+                print(f"Deleted asset: {os.path.basename(asset_path)}.")
+            else:
+                print(f"Failed to delete asset: {os.path.basename(asset_path)}. Status code: {response.status}")
+                return False
+    else:
+        print(f"Asset {os.path.basename(asset_path)} not found. Nothing to delete.")
+
+    upload_url = f"https://uploads.github.com/repos/{repo}/releases/{release_id}/assets?name={os.path.basename(asset_path)}"
+    async with aiofiles.open(asset_path, 'rb') as f:
+        data = await f.read()
+    async with session.post(upload_url, headers=headers, data=data) as response:
+        result = await response.json()
+    print(f"Upload completed for: {os.path.basename(asset_path)}.")
+    return result
+
+def setSDKToDeviceInDb(dbs, regex, ai_sdk):
+    for eachDb in dbs:
+        if eachDb:
+            updateTableCollumn(
+                eachDb,
+                "Devices",
+                "sdk_support",
+                1,
+                "uid",
+                regex
+            )
+            if ai_sdk:
+                query = '''UPDATE Devices SET sdk_config = REPLACE(sdk_config, '}', ',"AI_GENERATED_SDK":"True"}') WHERE uid REGEXP "''' + regex + '"'
+                updateTableCollumn(
+                    eachDb,
+                    None,
+                    None,
+                    None,
+                    None,
+                    None,
+                    query
+                    )
+
+
+    return
+
+def removeDeviceFromDb(dbs, regex):
```

Comment:
Add an option to keep the device in the database, just hide it in SDK.

Take care of special cases for XC compilers.

Reply:
Added a checkbox for XC8 support adding.